Features: theme and i18n

makcimishevskih opened the pull request.

foxtrotkilomike commented on public/locales/ru/orgPage.json:
```diff
@@ -0,0 +1,9 @@
+{
+  "title": "Список репозиториев организации",
+  "subtitle": "Список репозиториев для специфичных организаций",
```
Для специфичных организаций 🤣 🤣 🤣
Не, смысл вообще был не в этом) Типа "список репозиториев определенной организации"
Мне кажется вообще это бессмысленный подзаголовок, может уберем его?...

foxtrotkilomike commented on src/components/Header/Header.module.scss:
```diff
@@ -2,30 +2,43 @@
       height: 100%;
     }
+
+    &__left {
+      display: flex;
+      align-items: center;
+      gap: $space-xl;
+    }
+
+    &__right {
```
На будущее, класснеймы лучше не называть в привязке к расположению, т.к. структура может поменяться и они потеряют смысл. Нужно стараться называть согласно тому контенту, который внутри этих элементов находится

foxtrotkilomike commented on src/components/ThemeButton/ThemeButton.tsx:
```diff
@@ -0,0 +1,27 @@
+interface IThemeButtonProps {}
+
+const ThemeButton: FC<IThemeButtonProps> = () => {
+  const { theme, changeTheme } = useLocalStore<ThemeStore>(() => new ThemeStore());
```
Тут неверно выбрана семантика стора – тема же глобальна для всего приложения, информация о ней используется всеми компонентами, поэтому стор тоже должен быть глобальным

foxtrotkilomike commented on src/providers/RouterProvider/RouterProvider.tsx:
```diff
@@ -1,9 +1,9 @@
 
-import { BrowserRouter } from 'react-router-dom';
+import { HashRouter } from 'react-router-dom';
 
 const RouterProvider: FC<PropsWithChildren> = ({ children }) => {
-  return <BrowserRouter>{children}</BrowserRouter>;
+  return <HashRouter>{children}</HashRouter>;
```
А какова причина перехода на HashRouter? Чтобы поддержать прямой переход по ссылке в github pages? Его, вообще, используют в крайне редких случаях, когда других вариантов нет

foxtrotkilomike commented on src/store/ThemeStore/ThemeStore.ts:
```diff
@@ -0,0 +1,38 @@
+  };
+
+  changeTheme = () => {
+    this.theme === 'light' ? (this.theme = 'dark') : (this.theme = 'light');
```
Какой-то странный синтаксис :)
Обычно пишут так:
```
this.theme = this.theme === 'light' ? 'dark' : 'light';
```

foxtrotkilomike commented on src/styles/vars.scss:
```diff
@@ -1,63 +1,67 @@
+@import "./themeVars.scss";
+
+// sass
+$background-color: var(--background-color);
+$background-color-invert: var(--background-color-invert);
```
В целом посмотрел, как темы создавать, вот этот вариант, который ты сделал (с css-переменными в основе) – наиболее простой и понятный, поэтому тут всё ок. Есть ещё варианты с тем, чтобы генерировать эти css переменные через мапы и миксины в SASS, но это уже заморочки, суть не меняется.

foxtrotkilomike reviewed (commented):
В целом всё хорошо, вот ещё несколько общих комментариев:
- очень желательно оставлять описание пулл-реквеста – что было сделано/добавлено/улучшено по пунктам чеклистом с отмеченными пунктами (и с неотмеченными, если что-то не было реализовано). Если это визуальные стилевые улучшения, то можно приложить скриншот.
- довольно критично с точки зрения UX - когда ввожу какую-то другую организацию (не ktsstudio), потом перехожу в репозиторий и возвращаюсь обратно - результаты поиска сбрасываются и снова показываются репозитории kts. Нужно сохранять состояние поиска (мб в localstorage или в глобальном сторе), и восстанавливать его при возвращении на страницу
- если будешь дорабатывать для защиты - нужно добавить компонент-заглушку, если список репозиториев пустой (сейчас если выбрать приватные репозитории - ничего не покажет)
- я бы попробовал сделать автоматическое обновление результатов поиска при смене значения дропдауна, чтобы не пришлось отдельно нажимать кнопку поиска
- не понял, что за иконка в правом верхнем углу - при нажатии ничего не происходит. Надо либо оживить, либо убрать.
- на странице репозитория переход по ссылке в репу я бы делал с target="_blank"

makcimishevskih commented:
1. input and orgStore properly changes with localStorage.getItem('orgName')
 2. after change filter will send request
 3. added hook useFocus, refactor navInputs.
 4. refactored routes config and Link to
 5. Refactor orgReposList conditional rendering, and status messages

makcimishevskih commented:
1. fix: filters
2. fix: pagination conditional render
3. fix: json ru translations update to many stars
4. fix: createRoute deleted

makcimishevskih commented:
1. added: status component
2. fix: decor styles
3. added: node ./transformRuTranslations.js script

makcimishevskih commented:
1. delete: checkbox comp&styles, iconwrapper
2. fix: adaptive, ltl styles
3. masks to pathes

makcimishevskih commented:
1. add:contributors button to show-hide
2. refact:readme
3. fix:loader readme error
4. fix:some translate,animation
